### PR TITLE
[CGTN] Add new extractor

### PR DESCRIPTION
**Pull request by chao813**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Read [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site)
- [x] Read [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) and adjusted the code to meet them
- [x] Covered the code with tests (note that PRs without tests will be REJECTED)
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

#29157 Add site support for CGTN.

### CODE REVIEW

File: youtube_dl/extractor/cgtn.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        }
+    ]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        video_id = video_id.split('-')[-1]
+        webpage = self._download_webpage(url, video_id)
+
+        title = self._html_search_regex(r'<div class="news-title">(.+?)</div>', webpage, 'title')
+        download_url = self._html_search_regex(r'<div class="cg-player-container J_player_container" .*? data-video ="(?P<url>.+m3u8)" (.*?)>', webpage, 'download_url')
```
Comment (dstftw) [inline youtube_dl/extractor/cgtn.py:35]: Relax regex.

File: youtube_dl/extractor/cgtn.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        thumbnail = self._html_search_regex(r'<div class="cg-player-container J_player_container" .*? data-poster="(?P<thumbnail>.+jpg)" (.*?)>', webpage, 'thumbnail')
```
Comment (dstftw) [inline youtube_dl/extractor/cgtn.py:36]: Should not be fatal.

File: youtube_dl/extractor/cgtn.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                'ext': 'mp4',
+                'title': 'China, Indonesia vow to further deepen maritime cooperation',
+                'thumbnail': r're:^https?://.*\.jpg$',
+            }
+        }
+    ]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        video_id = video_id.split('-')[-1]
```
Comment (dstftw) [inline youtube_dl/extractor/cgtn.py:31]: Move this into `_VALID_URL`.

**Review by dstftw — CHANGES_REQUESTED** [review]

File: youtube_dl/extractor/cgtn.py (hunk 2 above)
Comment (chao813) [reply youtube_dl/extractor/cgtn.py:36]: What do you mean by should not be fatal?

**Review by chao813 — COMMENTED** [review]
Fixed changes requested by @dstftw

**Comment by chao813** [comment]
@dstftw Can this branch be merged? I fixed the changes you requested, am I missing anything else? Thanks!

**Comment by chao813** [comment]
@dstftw

**Pull request closed without merging**